### PR TITLE
Incremental scheduler changes in support of parallel scheduler

### PR DESCRIPTION
**Pull request by vaporos**

### CODE REVIEW

File: validator/sawtooth_validator/execution/scheduler_parallel.py [hunk 1]
```diff
@@ -77,64 +78,148 @@ def set_writer(self, address, writer):
...
+        """Returns all predecessor transaction ids for a write of the provided
+        address.
+
+        Arguments:
+            address (str): the radix address
+
+        Returns: a set of transaction ids
+        """
+        # A write operation must be preceded by:
+        #   - The "enclosing writer", which is th writer at the address or
```
Comment (jsmitchell) [inline validator/sawtooth_validator/execution/scheduler_parallel.py]: s/th/the/

File: validator/sawtooth_validator/execution/scheduler_parallel.py [hunk 2]
```diff
@@ -77,64 +78,148 @@ def set_writer(self, address, writer):
...
+        # not need to include them here as they will have been considered a
+        # predecessor to the enclosing writer.
+        #
+        # Enclosing readers must be included.  Technically, we only need to add
+        # enclosing readers which occurred after the enclosing writer, since
+        # the readers preceding the writer will have been considered a
+        # predecessor of the enclosing writer.  However, with the current
+        # data structure we can not determine the difference between readers
+        # so we specify them all; this is mostly harmless as it will not change
+        # the eventual topological sort order.
```
Comment (jsmitchell) [inline validator/sawtooth_validator/execution/scheduler_parallel.py]: Is this relevant per the removal of topological sort?

File: validator/sawtooth_validator/execution/scheduler_parallel.py (hunk 1 above)
Comment (vaporos) [inline validator/sawtooth_validator/execution/scheduler_parallel.py]: Fixed in new commits.

File: validator/sawtooth_validator/execution/scheduler_parallel.py (hunk 2 above)
Comment (vaporos) [inline validator/sawtooth_validator/execution/scheduler_parallel.py]: Fixed in new commits.

**Comment by vaporos** [comment]
Rebased without code modifications.

**Pull request merged**